File: redwood-yoga-java/src/main/java/io/github/orioncraftmc/meditate/interfaces/YogaLogger.java
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package io.github.orioncraftmc.meditate.interfaces;


import io.github.orioncraftmc.meditate.enums.YogaLogLevel;
import io.github.orioncraftmc.meditate.internal.YGConfig;
import io.github.orioncraftmc.meditate.internal.YGNode;
import io.github.orioncraftmc.meditate.internal.enums.YGLogLevel;
import io.github.orioncraftmc.meditate.internal.interfaces.YGLogger;

/**
 * Interface for receiving logs from native layer. Use by setting YogaNode.setLogger(myLogger);
 * See YogaLogLevel for the different log levels.
 */
public interface YogaLogger extends YGLogger {
    void log(YogaLogLevel level, String message);

    @Override
    default int invoke(YGConfig config, YGNode node, YGLogLevel level, String format, Object... args) {
        log(YogaLogLevel.fromInt(level.getValue()), String.format(format, args));
        return 0;
    }
}

File: redwood-yoga/src/commonMain/kotlin/app/cash/redwood/yoga/interfaces/YogaLogger.kt
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
package app.cash.redwood.yoga.interfaces

import app.cash.redwood.yoga.enums.YogaLogLevel
import app.cash.redwood.yoga.internal.YGConfig
import app.cash.redwood.yoga.internal.YGNode
import app.cash.redwood.yoga.internal.enums.YGLogLevel
import app.cash.redwood.yoga.internal.interfaces.YGLogger

/**
 * Interface for receiving logs from native layer. Use by setting YogaNode.setLogger(myLogger);
 * See YogaLogLevel for the different log levels.
 */
interface YogaLogger : YGLogger {
    fun log(level: YogaLogLevel?, message: String?)

    override fun invoke(
      config: YGConfig?,
      node: YGNode?,
      level: YGLogLevel,
      format: String,
      vararg args: Any?
    ): Int {
        // FIXME: Uncomment, port to common
        //log(YogaLogLevel.Companion.fromInt(level.getValue()), String.format(format!!, *args))
        return 0
    }
}
